Feature/429 이력서 작성 UI 개선

MrMirror21 opened the pull request:
## Related issue 🛠

- closed #429 

## Work Description ✏️
<div align="center"><img width="320" alt="스크린샷 2025-06-27 오전 11 41 12" src="https://github.com/user-attachments/assets/6f93530e-8651-46dd-b1ab-f52e68672d8d" />
<img width="325" alt="스크린샷 2025-06-27 오전 11 40 44" src="https://github.com/user-attachments/assets/85485974-67dc-40b1-b5be-d71a0dea9310" />
<img width="320" alt="스크린샷 2025-06-27 오전 11 40 02" src="https://github.com/user-attachments/assets/aa78e889-bd31-4505-abd8-595cf4b1ea74" /></div>

이력서 항목(자기소개, 학력, 경력)의 생성/수정 페이지 로직을 개선하고 코드 구조를 리팩토링했습니다.
- 각 페이지 UI 변경 적용
- 페이지 및 폼 컴포넌트 통합
  - 경력(`WorkExperience`)의 생성/수정 페이지를 각각 단일 페이지 컴포넌트(`WorkExperiencePage.tsx`)로 통합
  - 관련 폼 컴포넌트를 `WorkExperienceForm.tsx`으로 통합
- 커스텀 훅을 통한 로직 추상화
  - 페이지별 비즈니스 로직(상태 관리, API 연동, 핸들러 등)을 각각의 커스텀 훅(`useWorkExperience`, `useEducation`)으로 분리
- 유효성 검사 로직 개선
  - `utils/editResume.ts` 내 유효성 검사 로직을 강화
  - 중복되는 날짜 검증 로직을 `isValidDateString` 헬퍼 함수로 분리

## Uncompleted Tasks 😅

- [ ] #428 pr merge 후 Pagetitle 적용됐는지 화깅ㄴ

## To Reviewers 📢
- 학력 및 경력 페이지에 생성/수정 로직을 통합하고, 비즈니스 로직을 커스텀 훅(`useEducation`, `useWorkExperience`)으로 분리하는 리팩터링 패턴을 적용했습니다. 추후 한 페이지 내에서 상태 관리와 api 요청, 유효성 검사, 이벤트 핸들러 등이 복잡하게 얽힌 페이지의 경우 이 패턴을 참고하면 좋을 것 같습니다.
- 날짜 문자열의 유효성을 상세히 검증하는 `isValidDateString` 함수가 `utils/editResume.ts`에 추가되었습니다. 날짜 입력 폼 구현 시 활용하시면 됩니다.
- #428 PR에서 `PageTitle` 컴포넌트를 수정했는데 이곳에는 적용되지 않아 페이지 상단 padding이 과해보일 수 있는데 이 점 참고 부탁드립니다.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **신규 기능**
  * 학력 및 경력 입력/수정 페이지의 상태 관리와 제출 로직을 커스텀 훅으로 분리하여 일관성과 유지보수를 강화했습니다.
  * 경력 입력 페이지가 단일 컴포넌트로 통합되어 신규 및 수정 모두 지원합니다.
  * 학력, 경력 입력 폼에 입력값 유효성 검증 및 즉각적인 피드백 기능이 추가되었습니다.

* **버그 수정**
  * 경력 관련 타입 오타(WorkExperienctRequest → WorkExperienceRequest)를 수정하여 API 및 타입 일관성을 확보했습니다.

* **스타일**
  * 학력, 경력, 자기소개 등 주요 입력 폼의 여백, 정렬, 구분선 등 UI 레이아웃이 개선되어 가독성과 사용성이 향상되었습니다.
  * 입력란의 플레이스홀더, 안내문구, 버튼 스타일이 통일되고 명확해졌습니다.

* **리팩터링**
  * 학력 및 경력 입력 페이지의 상태 관리, API 호출, 유효성 검사 로직을 컴포넌트에서 커스텀 훅으로 분리하여 코드 구조를 개선했습니다.
  * 불필요한 중복 컴포넌트 삭제 및 라우팅 구조 단순화가 이루어졌습니다.

* **문서화**
  * 입력 폼의 안내 문구와 플레이스홀더가 실제 입력 목적에 맞게 더 명확하게 수정되었습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

MrMirror21 committed:
🎨 style: 자기 소개 작성 ux 개선 #429

- ui 변경 적용
- 자수 제한 표기 ui 변경

MrMirror21 committed:
🎨 style: 학력 입력 ux 개선 #429

- ui 변경 적용
- 학년, 학점 입력 시 입력 안정성 높이는 방향으로 수정

MrMirror21 committed:
♻️ refactor: 경력 입력/수정 페이지 통합 #429

- 분리되어있던 경력 입력/수정 페이지 및 컴포넌트를 WorkExperiencePage로 통합
- WorkExperiencePost/Patch 컴포넌트를 WorkExperienceForm으로 통합

MrMirror21 committed:
✨ feat: 날짜 입력 유효성 검사 로직 개선 #429

- 학력, 경력 입력 시 실제 날짜 여부와, 종료일자가 시작일자보다 미래임을 확인하도록 개선

MrMirror21 committed:
♻️ refactor: 날짜 형식 유효성 검사식 중복 제거 #429

MrMirror21 committed:
♻️ refactor: 경력/학력 입력/수정 페이지 로직 추상화 #429

- 경력/학력 입력/수정 페이지 내 상태 관리, API 호출, 유효성 검사, 이벤트 핸들러 등을 커스텀 훅으로 캡슐화
- 페이지 파일은 UI를 렌더링하는데 집중

MrMirror21 committed:
♻️ refactor: 학력 입력 UI 내 숫자 입력 핸들러 추상화 #429

- 기존 학점/학년 입력 핸들러를 각각의 경우로 분리

MrMirror21 committed:
🎨 style: 경력/학력 등 다중 요소 추가 시 ui 깨짐 수정 #429

MrMirror21 committed:
Merge branch 'dev' of https://github.com/Team-inglo/Giggle-Web into feature/429-이력서-작성-ui-개선

MrMirror21 committed:
🔧 chore: 타입 오타 수정 #429

MrMirror21 committed:
🔧 chore: 이전 PR 미적용 사항들 적용 #429

- 미사용 파일 (WorkExperiencePatch) 제거
- PageTitle 컴포넌트가 변경되면서 하위 요소 py 패딩 제거
- 라우터 파일 내 상대 경로 제거

naarang commented on src/hooks/useEducation.ts:
```diff
@@ -0,0 +1,132 @@
+    handleSubmit,
+    handleReset,
+  };
+};
```
훅으로 분리하는 아이디어는 정말 좋은 것 같습니다! 근데 return되는 값이 굉장히 많고 훅 로직 역시 여전히 비대하다는 점이 고민이 되네요ㅠ

훅을 다음처럼 관심사별로 관리해보는 것은 어떨까요??

```js
const { educationData, setEducationData, isValid, handleReset, pageTitle } =
  useEducationForm();

const { mode, isPending, initialData, schoolData, handleSubmit } =
  useEducationApi(educationData);
```

naarang reviewed (approved):
통합, 추상화로 리팩토링에 신경을 많이 써주셨네요!!👍 수고많으셨습니다

MrMirror21 committed:
Merge branch 'dev' of https://github.com/Team-inglo/Giggle-Web into feature/429-이력서-작성-ui-개선

MrMirror21 committed:
♻️ refactor: useEducation 훅 분리 #429

- form을 담당하는 useEducationForm.ts와 api를 담당하는 useEducationApi.ts로 분리
- 전공 enum관련 타입 매칭 이슈도 같이 해결

MrMirror21 committed:
🐛 fix: 학교 검색, 선택 ui 변경 적용, 동작 정상화 #429

- 누락됐던 학교 검색 ui 디자인 변경 적용
- 학교 검색 ui를 클릭해도 학교 검색 ui가 나오지 않던 이슈 해결
